tests/testthat/test-data-set.R

```diff
@@ -60,3 +59,58 @@ test_that("it can update the dimension of the yValues and this does not change t
   expect_equal(dataSet$yValues, c(1, 2, 4))
   expect_equal(dataSet$yUnit, ospUnits$Ampere$A)
 })
+
+test_that("it can set the xValues if xUnit is different from base unit", {
+  dataSet <- DataSet$new()
+  dataSet$xUnit <- ospUnits$Time$`week(s)`
+  dataSet$xValues <- c(1, 2, 3, 4, 5)
+  expect_equal(dataSet$xValues, c(1, 2, 3, 4, 5))
+  expect_equal(dataSet$xUnit, ospUnits$Time$`week(s)`)
+})
+
+test_that("it can set the yValues if yUnit is different from base unit", {
+  dataSet <- DataSet$new()
+  dataSet$xValues <- c(1, 2, 3, 4, 5)
+  dataSet$yUnit <- "mg/l"
+  dataSet$yValues <- c(1, 2, 3, 4, 5)
+  #TODO this test fails because of numerical error in the conversion if default tolerance is used
+  expect_equal(dataSet$yValues, c(1, 2, 3, 4, 5), tolerance = 0.0001)
+  expect_equal(dataSet$yUnit, "mg/l")
+})
+
+test_that("it can change yErrorType when no yError is defined", {
+  dataSet <- DataSet$new()
+  expect_equal(dataSet$yErrorType, DataErrorType$ArithmeticStdDev)
+  dataSet$yErrorType <- DataErrorType$GeometricStdDev
+

```

Comment:
Failing because getting yValues or yErrorValues does not work if the values are not set.